src/jit/lsra.cpp

```diff
@@ -5047,17 +5054,46 @@ void LinearScan::assignPhysReg( RegRecord * regRec, Interval * interval)
         }
         unassignPhysReg(regRec->regNum);
     }
+
     regRec->assignedInterval = interval;
+}
+
+// Assign the given physical register interval to the given interval
+void LinearScan::assignPhysReg( RegRecord * regRec, Interval * interval)
+{
+    regMaskTP assignedRegMask = genRegMask(regRec->regNum);
+    compiler->codeGen->regSet.rsSetRegsModified(assignedRegMask DEBUG_ARG(dumpTerse));
+
+    checkAndAssignInterval(regRec, interval);
     interval->assignedReg = regRec;
 
 #ifdef _TARGET_ARM_
     if ((interval->registerType == TYP_DOUBLE) && isFloatRegType(regRec->registerType))
     {
-        RegRecord * nextRegRec = getRegisterRecord(REG_NEXT(regRec->regNum));
-        nextRegRec->assignedInterval = interval;
+        regNumber   nextRegNum = REG_NEXT(regRec->regNum);
+        RegRecord * nextRegRec = getRegisterRecord(nextRegNum);
+
+        checkAndAssignInterval(nextRegRec, interval);
     }
 #endif // _TARGET_ARM_
 
+#if FEATURE_MULTIREG_ARGS_OR_RET
+    if (interval->registerType == TYP_STRUCT)
```

Comment:
Probably - Brian can answer better. I suspect that even 16-byte SIMD types will be passed in multiple registers.